sam.c

```diff
@@ -1097,13 +1102,13 @@ static bam_hdr_t *sam_hdr_sanitise(bam_hdr_t *h) {
     if (last != '\n') {
         hts_log_warning("Missing trailing newline on SAM header. Possibly truncated");
 
-        if (h->l_text == UINT32_MAX) {
-            hts_log_error("No room for extra newline");
-            bam_hdr_destroy(h);
-            return NULL;
-        }
+        if (h->l_text < 2 || i >= h->l_text - 2) {
+            if (h->l_text >= SIZE_MAX - 2) {
```

Comment:
SIZE_MAX again...

Reply:
Given this is a continual problem, maybe we should just fix the C++ brokenness.  Eg:

```
#ifndef SIZE_MAX
#  define SIZE_MAX ((size_t)-1)
#endif
```

Although signed overflow is undefined, promotion rules are different (I think!).  This is making assumption on 2s complement arithmetic, but that's an assumption we're safe with.